components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/DataServiceDocLitWrappedSchemaGenerator.java

```diff
@@ -91,7 +91,7 @@ public static void populateServiceSchema(AxisService axisService) throws DataSer
             processRequest(cparams, resource);
         }
         /* process requestBox operation (Only have one element) */
-        if (dataservice.isBoxcarringEnabled()) {
+        if (dataservice.isBoxcarringEnabled() && dataservice.isDisableLegacyBoxcarringMode()) {

```

Comment:
Noted. 

But aiya if customer use REQUEST BOX operation without disabling the legacy boxcarring mode, then again there is a chance to occur OOM. that's why I add this condition.